src/python_testing/TC_SU_2_2.py

```diff
@@ -0,0 +1,1251 @@
+            val = getattr(report.value, "value", report.value)  # unwrap Nullable if needed
+
+            if val == NullValue:
+                val = None
+
+            # Track progress numbers 1-99
+            if val is not None and isinstance(val, int) and 1 <= val <= 99:
+                progress_seen = True
+                if val not in progress_recorded:
+                    progress_recorded.add(val)
+                    progress_values.append(val)
+                    logger.info(f'{step_number}: 1.2.2 - UpdateStateProgress recorded: {val}')
+                return False
+
+            # Track final None after some progress
+            if val is None and progress_seen:
+                final_null_seen = True
+                progress_values.append(val)
+                logger.info(f'{step_number} 1.2.2 - OTA UpdateStateProgress sequence complete, final state is None: {progress_values}')
+                return final_null_seen
+
+            return False
+
+        # Create matcher object for UpdateStateProgress
+        matcher_progress_obj = AttributeMatcher.from_callable(
+            description="Validate OTA UpdateStateProgress transitions: non-null values rage 1-99 and final None",
+            matcher=matcher_progress
+        )
+
+        # ------------------------------------------------------------------------------------
+        # [STEP_1]: Step #1.3 - Start tasks to track OTA attributes:
+        # UpdateState and UpdateStateProgress (updateAvailable sequence) with validations
+        # ------------------------------------------------------------------------------------
+        try:
+            # Wait until the final state (Idle) is reached or timeout (5 min)
+
+            thread_state = threading.Thread(
+                target=lambda: subscription_attr_state.await_all_expected_report_matches(
+                    [matcher_update_state_obj], timeout_sec=300.0)
+            )
+            thread_progress = threading.Thread(
+                target=lambda: subscription_attr_progress.await_all_expected_report_matches(
+                    [matcher_progress_obj], timeout_sec=300.0)
+            )
+
+            # Start both threads
+            thread_state.start()
+            thread_progress.start()
+
+            # Wait for both threads to complete
+            thread_state.join()
+            thread_progress.join()
+
+            logger.info(
+                f'{step_number}: Step #1.3 - Both OTA attributes UpdateState (1.2.1) and UpdateStateProgress (1.2.2) matchers have completed.')
+        except Exception as e:
+            logger.warning(f'OTA update encountered an error or timeout: {e}')
+        finally:
+            # Cancel both subscriptions and task
+            await subscription_attr_state.cancel()
+            await subscription_attr_progress.cancel()
+
+        # ------------------------------------------------------------------------------------
```

Comment:
The provider stuff from Juan E has landed, so this can be tightened up now. For this step, just wait until you see SOME progress - ie download started, UpdateStateProgress it not null, then cancel the provider and wait for the device to go back to idle. You won't need to verify the image transfer.

The test plan will need updates, but it's minor.